kernel/src/metrics/events.rs

```diff
@@ -0,0 +1,195 @@
+//! Metric event types and utilities.
+
+use std::fmt;
+use std::time::{Duration, Instant};
+use uuid::Uuid;
+
+/// Unique identifier for a metrics operation.
+///
+/// Each operation (Snapshot, Transaction, Scan) gets a unique MetricId that
+/// is used to correlate all events from that operation.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub struct MetricId(Uuid);
+
+impl MetricId {
+    /// Generate a new unique MetricId.
+    pub fn new() -> Self {
+        Self(Uuid::new_v4())
+    }
+}
+
+impl Default for MetricId {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl fmt::Display for MetricId {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.0)
+    }
+}
+
+/// Metric events emitted during Delta Kernel operations.
+///
+/// Some events include an `operation_id` (MetricId) that uniquely identifies the operation
+/// instance. This allows correlating multiple events from the same operation.
+#[derive(Debug, Clone)]
+pub enum MetricEvent {
+    /// A snapshot creation operation has started.
```

Comment:
Do we need to know this? I'd like to match what the java kernel has so far, so let's make sure we're not adding anything they don't have just yet